lib/vsc/utils/generaloption.py

```diff
@@ -1669,11 +1678,18 @@ def generate_cmd_line(self, ignore=None, add_default=None):
                             opt_value = opt_value + -default
                         else:
                             opt_value = -default + opt_value
-                    elif hasattr(opt_value, '__getslice__'):
-                        if action == 'add_first':
-                            opt_value = opt_value[:-len(default)]
-                        else:
-                            opt_value = opt_value[len(default):]
+                    # __getslice__ no longer exists in Python 3,
+                    # and just having __getitem__ is not sufficient to support slicing;
+                    # so, just try slicing, and ignore if it fails
+                    elif hasattr(opt_value, '__getslice__') or hasattr(opt_value, '__getitem__'):
+                        try:
+                            if action == 'add_first':
+                                opt_value = opt_value[:-len(default)]
+                            else:
+                                opt_value = opt_value[len(default):]
+                        except TypeError:
+                            # if slicing fails, a TypeError "is not subscriptable" will be raised
```

Comment:
log an error at the very least (or remove the try/except). why would it be ok to simply ignore this?

Reply:
I've removed the try/except. If we trigger an exception here, the code that uses `generaloption` is doing something weird/wrong, so it's OK to let this fail hard in that case